src/components/user-list.js

```diff
@@ -2,51 +2,85 @@ import React from 'react';
 
 import '../css/app.css';
 import UserStore from '../stores/user-store';
-import CHANGE_EVENT from '../stores/event-type';
+import EVENT_TYPE from '../stores/event-type';
 
 import User from './user.js';
+import FindUser from './find-user';
 
 export default class UserList extends React.Component {
     constructor (props) {
         super(props);
 
         this.state = this._getAppState();
 
-        this.handleClickAddUser = this.handleClickAddUser.bind(this);
+        this._handleClickAddUser = this._handleClickAddUser.bind(this);
+        this._handleClickFindUser = this._handleClickFindUser.bind(this);
     }
 
     _getAppState () {
         return {
             wantAdd:  false,
             wantEdit: false,
+            wantFind: false,
             users:    UserStore.getUsers()
         };
     }
 
-    handleClickAddUser () {
+    _handleClickAddUser () {
         this.setState({ wantAdd: !this.state.wantAdd });
     }
 
-    _onChange = () => {
-        this.setState(this._getAppState());
+    _onChange = (userList) => {
+        this.setState({ users: userList });
     };
 
+    _handleClickFindUser () {
+        const { wantFind } = this.state;
+
+        if (wantFind) {
+            this.setState({ users: UserStore.getUsers() });
+        }
+        this.setState({ wantFind: !wantFind });
+    }
+
     componentDidMount () {
-        UserStore.on(CHANGE_EVENT, this._onChange);
+        UserStore.on(EVENT_TYPE.change, this._onChange);
     }
 
     componentWillUnmount () {
-        UserStore.removeListener(CHANGE_EVENT, this._onChange);
+        UserStore.removeListener(EVENT_TYPE.change, this._onChange);
     }
 
     render () {
-        const { wantAdd, users } = this.state;
+        const { wantAdd, users, wantFind } = this.state;
 
         return (
             <div className="UserList">
-                <button className="ButtonAddUser" data-testid="ButtonAddUser" onClick={this.handleClickAddUser}>
-                    Add new User
+
+                <button className="ButtonFindUser" onClick={this._handleClickFindUser}>
+                    {wantFind ? 'Stop searching' : 'Find User'}
                 </button>
+                {
+                    wantFind &&
+                    (
+                        <FindUser/>
+                    )
+                }
+                {
+                    !wantFind &&
+                    (
+                        <button className="ButtonAddUser" data-testid="ButtonAddUser"
+                                onClick={this._handleClickAddUser}>
+                            Add new User
```

Comment:
Let's change name of this button if form is shown to e.g. `Cancel` or `Cancel additing`